prow/istio-pilot-multicluster-e2e.sh

```diff
@@ -14,14 +14,120 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# set this to the new multicluster-e2e type
-export RESOURCE_TYPE="${RESOURCE_TYPE:-gke-e2e-test}"
-export OWNER="istio-pilot-multicluster-e2e"
+set -eux
 
-export SETUP_CLUSTERREG="True"
-CLUSTERREG_DIR="${CLUSTERREG_DIR:-$(mktemp -d /tmp/clusterregXXX)}"
-export CLUSTERREG_DIR
+set -eux
 
-echo 'Running pilot multi-cluster e2e tests'
-date
-./prow/e2e-suite.sh --timeout 90m --cluster_registry_dir="$CLUSTERREG_DIR" --single_test e2e_pilotv2_v1alpha3 "$@"
+function setup_clusterreg () {
+    MAIN_CONFIG=""
+    for context in "${CLUSTERREG_DIR}"/*; do
+        if [[ -z "${MAIN_CONFIG}" ]]; then
+            MAIN_CONFIG="${context}"
+        fi
+        export KUBECONFIG="${context}"
+        kubectl delete ns istio-system-multi --ignore-not-found
+        kubectl delete clusterrolebinding istio-multi-test --ignore-not-found
+        kubectl create ns istio-system-multi
+        kubectl create sa istio-multi-test -n istio-system-multi
+        kubectl create clusterrolebinding istio-multi-test --clusterrole=cluster-admin --serviceaccount=istio-system-multi:istio-multi-test
+        CLUSTER_NAME=$(kubectl config view --minify=true -o "jsonpath={.clusters[].name}")
+        gen_kubeconf_from_sa istio-multi-test "${context}"
+    done
+    export KUBECONFIG="${MAIN_CONFIG}"
+}
+
+function gen_kubeconf_from_sa () {
+    local service_account=$1
+    local filename=$2
+
+    SERVER=$(kubectl config view --minify=true -o "jsonpath={.clusters[].cluster.server}")
+    SECRET_NAME=$(kubectl get sa "${service_account}" -n istio-system-multi -o jsonpath='{.secrets[].name}')
+    CA_DATA=$(kubectl get secret "${SECRET_NAME}" -n istio-system-multi -o "jsonpath={.data['ca\\.crt']}")
+    TOKEN=$(kubectl get secret "${SECRET_NAME}" -n istio-system-multi -o "jsonpath={.data['token']}" | base64 --decode)
+
+    cat <<EOF > "${filename}"
+      apiVersion: v1
+      clusters:
+         - cluster:
+             certificate-authority-data: ${CA_DATA}
+             server: ${SERVER}
+           name: ${CLUSTER_NAME}
+      contexts:
+         - context:
+             cluster: ${CLUSTER_NAME}
+             user: ${CLUSTER_NAME}
+           name: ${CLUSTER_NAME}
+      current-context: ${CLUSTER_NAME}
+      kind: Config
+      preferences: {}
+      users:
+         - name: ${CLUSTER_NAME}
+           user:
+             token: ${TOKEN}
+EOF
+}
+
+SUBNET_A="10.10.0.0/16"
+SUBNET_B="10.20.0.0/16"
+
+CONFIG_A=$(cat <<EOF
+kind: Cluster
+apiVersion: kind.sigs.k8s.io/v1alpha3
+networking:
+  podSubnet: ${SUBNET_A}
+  serviceSubnet: 10.255.10.0/24
+EOF
+)
+CONFIG_B=$(cat <<EOF
+kind: Cluster
+apiVersion: kind.sigs.k8s.io/v1alpha3
+networking:
+  podSubnet: ${SUBNET_B}
+  serviceSubnet: 10.255.10.0/24
+EOF
+)
+
+function cleanup_kind_cluster() {
+  echo "Test exited with exit code $?."
+  kind export logs --name "${1}" "${ARTIFACTS}/kind-${1}/"
```

Comment:
this fails at locally if `ARTIFACTS`  is not set. I assume it's some prow bash library set it up. and left the cluster undeleted.